### ISSUE TITLE
`Cannot find file at xxx although it was listed by readdir` after deleting folder from project registry

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#1134]
### Code to reproduce the issue:

```
npm init -y
pnpm i next
rm -rf node_modules/.registry.npmjs.org/next
pnpm i next
```

### Expected behavior:

Reinstalls package.

### Actual behavior:

```
Already up-to-date
 WARN  Cannot find file at /Users/Vaughan/dev-scratch/000000000/node_modules/next although it was listed by readdir
```

```
cat node_modules/.registry.npmjs.org/next
# cat: node_modules/.registry.npmjs.org/next: No such file or directory
```

Package is not reinstalled.

### Workaround

```
rm node_modules/next
pnpm i next

Already up-to-date
 WARN  Cannot find file at /Users/Vaughan/dev-quantitec/dev-intranav/intranav/node_modules/next although it was listed by readdir
 WARN  Cannot find file at /Users/Vaughan/dev-quantitec/dev-intranav/intranav/node_modules/@quantitec/babel-preset-intranav although it was listed by readdir

dependencies:
+ next 5.1.0 (6.0.0 is available)
```

But notice there is still the message that it cannot find `node_modules/next` even after I have deleted it.

### Additional information:

 - `node -v` prints: 8.1.2
 - Windows, OS X, or Linux?: macOS

**Comment by zkochan** [comment]
Why didn't you use `pnpm uninstall next`?

**Comment by vjpr** [comment]
The real issue was a sub-dependency that somehow didn't install its dependencies properly (I am tracking down this as a separate issue).

I came across this while searching for the root cause.

**Comment by cdfa** [comment]
Any progress on this? I've encountered the issue in 2 of my projects now, even though everything still seems to be working fine.

**Comment by melissamcewen** [comment]
I just encountered this error myself (I work support sometimes at Glitch.com). The user was getting
`WARN  Cannot find file at /rbd/pnpm-volume/....node_modules/consty although it was listed by readdir`. Weirdly consty does not seem to be a real module anywhere? It was listed as an Express dependency in shrinkwrap. I tried uninstalling express, downgrading Express. I could not get rid of it. Finally I installed a previous version of Express then ran `rm shrinkwrap.yaml && enable-pnp` and the shrinkwrap.yaml that was built doesn't seem to have `consty`.
